components/kyma-adapter/internal/handler/handler.go

```diff
@@ -1,16 +1,300 @@
+				respondWithError(ctx, w, http.StatusBadRequest, createErrorState, errors.Wrapf(err, fmt.Sprintf("while creating bundle instance auth for bundle with id: %q", bundle.ID)))
+				return
+			}
+		}
+	}
+
+	respondWithSuccess(ctx, w, readyState, fmt.Sprintf("Successfully created the bundle instance auths for runtime with ID %q", rtmID))
+}
+
+func (a AdapterHandler) processAuthRotation(ctx context.Context, w http.ResponseWriter, bundles []*graphql.BundleExt, reqBody types.Body, tenant string) {
+	rtmID := reqBody.GetRuntimeID()
+	basicCreds := reqBody.GetBasicCredentials()
+	oauthCreds := reqBody.GetOauthCredentials()
+
+	for _, bundle := range bundles {
+		for _, instanceAuth := range bundle.InstanceAuths {
+			if *instanceAuth.RuntimeID == rtmID {
+				if basicCreds.Username != "" { // basic creds
+					if err := a.DirectorGqlClient.UpdateBasicBundleInstanceAuth(ctx, tenant, instanceAuth.ID, bundle.ID, basicCreds.Username, basicCreds.Password); err != nil {
+						respondWithError(ctx, w, http.StatusBadRequest, createErrorState, errors.Wrapf(err, fmt.Sprintf("while updating bundle instance auth with id: %q", instanceAuth.ID)))
+						return
+					}
+				} else { // oauth creds
+					if err := a.DirectorGqlClient.UpdateOauthBundleInstanceAuth(ctx, tenant, instanceAuth.ID, bundle.ID, oauthCreds.TokenServiceURL, oauthCreds.ClientID, oauthCreds.ClientSecret); err != nil {
+						respondWithError(ctx, w, http.StatusBadRequest, createErrorState, errors.Wrapf(err, fmt.Sprintf("while updating bundle instance auth with id: %q", instanceAuth.ID)))
+						return
+					}
+				}
+			}
+		}
+	}
+
+	respondWithSuccess(ctx, w, readyState, fmt.Sprintf("Successfully updated the bundle instance auths for runtime with ID %q", rtmID))
+}
+
+type malformedRequest struct {
+	status int
+	msg    string
+}
+
+func (mr *malformedRequest) Error() string {
+	return mr.msg
+}
+
+// SuccessResponse structure used for JSON encoded success response
+type SuccessResponse struct {
+	State string `json:"state,omitempty"`
+}
+
+// ErrorResponse structure used for JSON encoded error response
+type ErrorResponse struct {
+	State   string `json:"state,omitempty"`
+	Message string `json:"error"`
+}
+
+func decodeJSONBody(w http.ResponseWriter, r *http.Request, dst interface{}) error {
+	if r.Header.Get(httputils.HeaderContentTypeKey) != "" {
+		if value, _ := header.ParseValueAndParams(r.Header, httputils.HeaderContentTypeKey); value != httputils.ContentTypeApplicationJSON {
+			return &malformedRequest{status: http.StatusUnsupportedMediaType, msg: "Content-Type header is not application/json"}
+		}
+	}
+
+	// Use http.MaxBytesReader to enforce a maximum read of 1MB from the
+	// response body. A request body larger than that will now result in
+	// Decode() returning a "http: request body too large" error.
+	r.Body = http.MaxBytesReader(w, r.Body, 1048576)
```

Comment:
Why do we chech for the size of the request here? we have an envoy filter that checks for the request size and denyes requests bigger than 2MB

Reply:
I didn't know about that envoy filter. I'll remove it.